(Conditionally) remove HTML from event description

Nico-J opened the pull request:
Good evening,

Earlier, we were encountering the problem that users of certain calendar tools, such as Apple Calendar, were seeing raw HTML tags in their event description, because obviously HTML seems not to be supported in those tools.

**So just remove that HTML stuff, replace `<br />` with `\n` and we're good to go, huh?**

Well, then the web version of Outlook Calendar doesn't support newline characters. It just strips them out. **Even the Windows-related `\r` newline character.**
It looks like this:
![image](https://user-images.githubusercontent.com/14024119/126570005-afdd0c0b-e348-4e39-a78b-84f14ffeaf9a.png)

Meanwhile, Google Calendar pretty much likes newline characters in plain text events:
![image](https://user-images.githubusercontent.com/14024119/126570089-73301ad5-6703-4cc9-910c-bd6b462caf84.png)

**So what do we do?**
I decided to conditionally use either the HTML-based version (for Outlook events) or the newline-based version (for everything else) of the event text when creating it. See the code (line 511 and following in calendarClient.ts) for more information.

Works in Outlook:
![image](https://user-images.githubusercontent.com/14024119/126570742-9a5eced5-a23c-440e-a6a3-6072485a865d.png)

And in Google:
![image](https://user-images.githubusercontent.com/14024119/126570775-fb71ccc4-f114-41f7-83e2-08607a90982f.png)

**But there's still an unsolved problem:**
Apple Calendar is just a tool that uses other integrations, such as Google or Outlook Calendar (or iCloud). So this PR doesn't solve the Apple Calendar issues, since we don't know which software the user is using.

We have the following options:
1. Just **always** throw away the HTML, leading to a quite "ugly" event description in Outlook Web, but saving the view for Apple Calendar.
2. Keep the conditional HTML usage: @Malte-D Could you test this PR with your Apple Calendar? I don't have a Mac right now. Maybe Outlook parses the HTML correctly and delivers a "working" version to Apple Calendar? Needs investigation!
3. Make an "opt-in" to let the user decide whether to use HTML descriptions or not.

Right now, this PR contains option 2. Please share your opinion about that with me. :)

Nico-J committed:
No HTMl in rich event description

Nico-J committed:
Properly replace a link tags

Nico-J committed:
Use regex to remove br tag

Nico-J committed:
Conditionally use HTML

Nico-J committed:
Added condition when updating event

baileypumfleet commented:
This code switches between HTML and plaintext based on the organiser's calendar integration (if it's Outlook or Google Calendar), however the invitee's calendar may or may not support HTML. For instance, if you on Google Calendar send me an invite with HTML, as I'm on Apple Calendar, it won't work for me. I think we may have to just send plaintext invites only. Great work though with this PR :)

PeerRich commented:
Until further I think we should do plaintext. No switches, no settings. Normal line breaks work for everyone right?

PeerRich reviewed (changes requested):
for now, plaintext only with line-breaks. great research and PR tho. these calendars are messy

Nico-J committed:
Always use plain text event descriptions for now

Nico-J commented:
Alright, I've changed the code to always use plain text now. In addition, I've added another RegEx to correctly strip down <a href="mailto:....">..</a> links. Before that, the "mailto:" part was kept there.

PeerRich commented:
@baileypumfleet can you run, review, then merge? :)
